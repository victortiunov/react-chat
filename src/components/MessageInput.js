import React from 'react';
import { withStyles } from 'material-ui/styles';
import { Input, Grid, Button } from 'material-ui';
import SendIcon from 'material-ui-icons/Send';
import Paper from 'material-ui/Paper';

const styles = theme => ({
  messageInputWrapper: {
    position: 'fixed',
    left: 'auto',
    right: 0,
    bottom: 0,
    width: 'calc(100% - 320px)',
  },
  messageForm: {
    padding: theme.spacing.unit,
  },
  messageInput: {
    marginTop: 15,
  },
  joinPaper: {
    padding: theme.spacing.unit * 2,
  },
});

class MessageInput extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      message: '',
    };
  }

  handleMessageChange = (e) => {
    this.setState({ message: e.target.value });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    if (this.state.message.trim()) {
      this.props.sendMessage(this.state.message);
      this.setState({ message: '' });
    }
  };

  render() {
    const {
      classes, user, onJoinClick, disabled,
    } = this.props;
    const { message } = this.state;

    return (
      <div className={classes.messageInputWrapper}>
        {user.isChatMember ? (
          <form className={classes.messageForm} onSubmit={this.handleSubmit}>
            <Grid container justify="center">
              <Grid item xs={8}>
                <Input
                  className={classes.messageInput}
                  fullWidth
                  value={message}
                  placeholder="Type your messageâ€¦"
                  onChange={this.handleMessageChange}
                  disabled={disabled}
                />
              </Grid>
              <Grid item className={classes.messageButton}>
                <Button
                  className={classes.messageButton}
                  type="submit"
                  variant="fab"
                  color="primary"
                  disabled={disabled || !message.trim()}
                >
                  <SendIcon />
                </Button>
              </Grid>
            </Grid>
          </form>
        ) : (
          <Paper className={classes.joinPaper} elevation={6}>
            <Button
              fullWidth
              variant="raised"
              color="primary"
              onClick={onJoinClick}
              disabled={disabled}
            >
              Join the Chat to start messaging
            </Button>
          </Paper>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(MessageInput);
