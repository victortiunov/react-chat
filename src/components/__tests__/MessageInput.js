/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from '../MessageInput';

const mockProps = {
  user: {
    isChatMember: false,
  },
  disabled: false,
  onJoinClick: jest.fn(),
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders join button', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled join button', () => {
    const tree = renderer.create(<MessageInput {...mockProps} disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders new message form', () => {
    const tree = renderer.create(<MessageInput {...mockProps} showJoinButton />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
