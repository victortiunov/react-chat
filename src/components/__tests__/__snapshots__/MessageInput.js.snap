// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders disabled join button 1`] = `
<div
  className="MessageInput-messageInputWrapper-57"
>
  <div
    className="MuiPaper-root-61 MuiPaper-elevation6-69 MuiPaper-rounded-62 MessageInput-joinPaper-60"
  >
    <button
      className="MuiButtonBase-root-103 MuiButtonBase-disabled-104 MuiButton-root-88 MuiButton-raised-93 MuiButton-raisedPrimary-95 MuiButton-disabled-97 MuiButton-fullWidth-102"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiButton-label-89"
      >
        Join the Chat to start messaging
      </span>
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders join button 1`] = `
<div
  className="MessageInput-messageInputWrapper-57"
>
  <div
    className="MuiPaper-root-61 MuiPaper-elevation6-69 MuiPaper-rounded-62 MessageInput-joinPaper-60"
  >
    <button
      className="MuiButtonBase-root-103 MuiButton-root-88 MuiButton-raised-93 MuiButton-raisedPrimary-95 MuiButton-fullWidth-102"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-89"
      >
        Join the Chat to start messaging
      </span>
      <span
        className="MuiTouchRipple-root-106"
      />
    </button>
  </div>
</div>
`;

exports[`<MessageInput /> renders new message form 1`] = `
<div
  className="MessageInput-messageInputWrapper-57"
>
  <div
    className="MuiPaper-root-61 MuiPaper-elevation6-69 MuiPaper-rounded-62 MessageInput-joinPaper-60"
  >
    <button
      className="MuiButtonBase-root-103 MuiButton-root-88 MuiButton-raised-93 MuiButton-raisedPrimary-95 MuiButton-fullWidth-102"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-89"
      >
        Join the Chat to start messaging
      </span>
      <span
        className="MuiTouchRipple-root-106"
      />
    </button>
  </div>
</div>
`;
